airflow/providers/amazon/aws/secrets/secrets_manager.py

```diff
@@ -96,28 +97,74 @@ def __init__(
 
     @cached_property
     def client(self):
-        """Create a Secrets Manager client"""
+        """
+        Create a Secrets Manager client
+        """
         session = boto3.session.Session(
-            profile_name=self.profile_name,
+            profile_name=self.profile_name
         )
         return session.client(service_name="secretsmanager", **self.kwargs)
 
-    def get_conn_uri(self, conn_id: str) -> Optional[str]:
+    def _get_extra(self, secret, conn_string):
+        if 'extra' in secret:
+            extra_dict = secret['extra']
+            kvs = "&".join([f"{key}={value}" for key, value in extra_dict.items()])
```

Comment:
What do you think about [urllib.parse.urlencode(](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode). I'm afraid the current implementation may have problems with some special characters.